File: benchmarks/api-nest/src/persona/persona.controller.ts
import { Controller, Get, Post, Body, Patch, Param, Delete, Logger } from '@nestjs/common';
import { PersonaService } from './persona.service';
import { CreatePersonaDto } from './dto/create-persona.dto';
import { UpdatePersonaDto } from './dto/update-persona.dto';

@Controller('persona')
export class PersonaController {
  private readonly logger = new Logger(PersonaController.name);
  constructor(private readonly personaService: PersonaService) {}

  @Post()
  create(@Body() createPersonaDto: CreatePersonaDto) {
    this.logger.log(`Creating persona with data: ${JSON.stringify(createPersonaDto)}`);
    return this.personaService.create(createPersonaDto);
  }

  @Get()
  findAll() {
    this.logger.log('Finding all personas');
    return this.personaService.findAll();
  }

  @Get(':id')
  findOne(@Param('id') id: string) {
    this.logger.log(`Finding persona with id: ${id}`);
    return this.personaService.findOne(id);
  }

  @Patch(':id')
  update(@Param('id') id: string, @Body() updatePersonaDto: UpdatePersonaDto) {
    this.logger.log(`Updating persona with id: ${id} and data: ${JSON.stringify(updatePersonaDto)}`);
    return this.personaService.update(id, updatePersonaDto);
  }

  @Delete(':id')
  remove(@Param('id') id: string) {
    this.logger.log(`Removing persona with id: ${id}`);
    return this.personaService.remove(id);
  }
}

File: benchmarks/api-c/src/persona/persona.controller.c
#include "persona.service.c"

// @GET: /persona
// @Inject: PersonaService
NestResponse persona_find_all_handler(PersonaService* s) {
    if (!s) return NESTC_ERROR("{\"error\": \"Service unavailable\"}");
    return NESTC_OK_T(s->find_all());
}

// @GET: /persona/:id
// @Inject: PersonaService
NestResponse persona_find_one_handler(PersonaService* s, const char* id) {
    if (!s) return NESTC_ERROR("{\"error\": \"Service unavailable\"}");
    return NESTC_OK_T(s->find_one(id));
}

// @POST: /persona
// @Inject: PersonaService
// @Body: CreatePersonaDto
NestResponse persona_create_handler(PersonaService* s, CreatePersonaDto* dto) {
    if (!s) return NESTC_ERROR("{\"error\": \"Service unavailable\"}");
    return NESTC_CREATED_T(s->create(dto));
}

// @PUT: /persona/:id
// @Inject: PersonaService
// @Body: UpdatePersonaDto
NestResponse persona_update_handler(PersonaService* s, const char* id, UpdatePersonaDto* dto) {
    if (!s) return NESTC_ERROR("{\"error\": \"Service unavailable\"}");
    return NESTC_OK_T(s->update(id, dto));
}

// @DELETE: /persona/:id
// @Inject: PersonaService
NestResponse persona_remove_handler(PersonaService* s, const char* id) {
    if (!s) return NESTC_ERROR("{\"error\": \"Service unavailable\"}");
    return NESTC_OK_T(s->remove(id));
}

